storage-access.bs

```diff
@@ -65,53 +69,84 @@ This specification depends on the Infra standard. [[!INFRA]]
+When an [=agent cluster=] is created, its [=agent cluster/storage access map=] is initialized with a [=map/clone=] of the [=global storage access map=].
 
-: The <dfn for="storage access flag set" id=has-storage-access-flag>has storage access flag</dfn>
-:: When set, this flag indicates |site| has access to its [=first party data=] when it's a [=third party=] on |top-level site|.
-: The <dfn for="storage access flag set" id=was-expressly-denied-storage-access-flag>was expressly denied storage access flag</dfn>
-:: When set, this flag indicates that the user expressly denied |site| access to its [=first party data=] when it's a [=third party=] on |top-level site|.
+To <dfn type="abstract-op">obtain the storage access map</dfn> for a {{Document}} |doc|, run the following steps:
+
+1. If |doc|'s [=Document/browsing context=] is not a [=top-level browsing context=], set |doc| to the [=active document=] of |doc|'s [=Document/browsing context=]'s [=top-level browsing context=].
+1. Return the [=agent cluster/storage access map=] of |doc|'s [=relevant agent=]'s [=agent cluster=].
+
+A <dfn>partitioned storage key</dfn> is a [=tuple=] consisting of a <dfn for="partitioned storage key">top-level site</dfn> and an <dfn for="partitioned storage key">embedded site</dfn> (both [=sites=]).
+
+<div class=example>
+
+`(("https", "news.example"), ("https", "social.example"))` is a [=partitioned storage key=] whose [=top-level site=] is `("https", "news.example")` and whose [=embedded site=] is `("https", "social.example")`.
+
+</div>
 
-To <dfn type="abstract-op">obtain the storage access flag set</dfn> for a {{Document}} |doc|, run the following steps:
+To <dfn type="abstract-op">generate a partitioned storage key</dfn> for a {{Document}} |doc|, run the following steps:
 
+1. If |doc|'s [=Document/browsing context=]'s [=opener browsing context=] is null, return failure.
+1. If |doc|'s [=Document/browsing context=]'s [=opener browsing context=] is not its [=top-level browsing context=], return failure.
 1. Let |settings| be |doc|'s [=relevant settings object=].
-1. Let |site| be the result of [=obtain a site|obtaining a site=] from |settings|' [=environment settings object/origin=].
+1. Let |embedded site| be the result of [=obtain a site|obtaining a site=] from |settings|' [=environment settings object/origin=].
 1. Let |top-level site| be the result of [=obtain a site|obtaining a site=] from |settings|' [=top-level origin=].
-1. Let |key| be the [=pair=] |top-level site|/|site|.
-1. Let |flags| be null.
-1. If [=storage access map=][|key|] [=exists=], set |flags| to [=storage access map=][|key|].
-1. If |flags| is null, run these steps:
-    1. Set |flags| to be a new [=storage access flag set=].
-    1. [=Set=] [=storage access map=][|key|] to |flags|.
-1. Return |flags|.
+1. Return the [=partitioned storage key=] (|top-level site|, |embedded site|).
+
+ISSUE(10): This algorithm intentionally fails for top frames and for nested <{iframe}>s.
```

Comment:
Why though? It seems to me it would be simpler to generate the same kind key for the top-level documents as well.

Reply:
Okay, that works for me. Fixed in 41a35bc.